### DISCUSSION TITLE
Does the Airsim/UE modeling of wind interaction include the effects of fluid dynamics on the propellers?

### DISCUSSION THREAD

**Discussion by m1baldwin** [microsoft/AirSim#3576] [Support Q&A]
Based on my understanding, it is currently possible to simulate wind within airsim by specifying certain parameters in settings.json, or by controlling the environment through the available APIs. 

When wind is introduced to the simulation, how is it actually simulated? And what level of interaction with the UAV is modeled? I'm not sure if my current understanding is correct; I believe that just the forces which would be acting on the body due to lift/drag would be modeled, but not turbulence, smooth airflow, etc. and their effects on the propellers. 

Can anyone help clear this up?

**Comment by m1baldwin** [comment]
I am basing my current understanding mainly off the PR here: https://github.com/microsoft/AirSim/pull/2867

**Comment by zimmy87** [comment]
Yes, your suspicion is correct. From looking at how wind is implemented in FastPhysicsEngine::getDragWrench(), the wind vector is applied in the calculation of drag on the PhysicsBody of the drone. Since there is a 1:1 relationship between the PhysicsBody instances and drones, this drag is only calculated for the whole of the drone's body. We do not currently model the propellers in FastPhysics as PhysicsBody's themselves, and I don't see any other places in the codebase where the wind vector is being read, so there is no calculation of the wind's effects on propellers as of yet.

**Comment by m1baldwin** [comment]
thanks @zimmy87 , I appreciate the second set of eyes on this
